azurerm/helpers/azure/app_service.go

```diff
@@ -159,8 +159,22 @@ func SchemaAppServiceSiteConfig() *schema.Schema {
 					Optional: true,
 					Default:  string(web.ScmTypeNone),
 					ValidateFunc: validation.StringInSlice([]string{
-						string(web.ScmTypeNone),
+						string(web.ScmTypeBitbucketGit),
+						string(web.ScmTypeBitbucketHg),
+						string(web.ScmTypeCodePlexGit),
+						string(web.ScmTypeCodePlexHg),
+						string(web.ScmTypeDropbox),
+						string(web.ScmTypeExternalGit),
+						string(web.ScmTypeExternalHg),
+						string(web.ScmTypeGitHub),
 						string(web.ScmTypeLocalGit),
+						string(web.ScmTypeNone),
+						string(web.ScmTypeOneDrive),
+						string(web.ScmTypeTfs),
+						string(web.ScmTypeVSO),
+						// Not in the specs, but is set by Azure Pipelines
+						// https://github.com/Microsoft/azure-pipelines-tasks/blob/master/Tasks/AzureRmWebAppDeploymentV4/operations/AzureAppServiceUtility.ts#L19
+						"VSTSRM",
```

Comment:
@hbuckle heads up that I've opened an issue on the Swagger repository to get this added to the specs: https://github.com/Azure/azure-rest-api-specs/issues/5345

Reply:
it'd be good to add a TODO to that issue here ^